torch/nn/utils/clip_grad.py

```diff
@@ -9,6 +10,18 @@
 
 __all__ = ['clip_grad_norm_', 'clip_grad_norm', 'clip_grad_value_']
 
+def _no_grad(func):
+    """
+    This wrapper is needed to avoid a circular import when using @torch.no_grad on the exposed functions
+    clip_grad_norm_ and clip_grad_value_ themselves.
+    """
+    def _no_grad_wrapper(*args, **kwargs):
+        with torch.no_grad():
+            return func(*args, **kwargs)
+    functools.update_wrapper(_no_grad_wrapper, func)
```

Comment:
```suggestion
    functools.update_wrapper(torch.no_grad(), func)
```
Doesn't this work? no_grad is already a contextlib wrapper I think

Reply:
Then I have to return torch.no_grad which still leads to a circular import

Reply:
What about 
```suggestion
    with torch.no_grad():
         yield
```

Reply:
I think you may be able to use yield to get out of this, but I am not sure...